### DISCUSSION TITLE
Better name for reconcileAt annotation

### DISCUSSION THREAD

**Discussion by squaremo** [fluxcd/flux2#363] [Proposals] [answered]
The annotation used by many GOTK types to trigger a reconciliation outside the specified schedule is called `fluxcd.io/reconcileAt`. I believe the idea was to imitate the annotation used in core Kubernetes, "restartedAt": https://github.com/kubernetes/kubectl/blob/master/pkg/polymorphichelpers/objectrestarter.go#L41. This works by changing the pod template of a workload, causing the controller to recreate all the pods. Any change will do there -- kubectl uses a timestamp, presumably so there's some indication what happened and when.

`reconcileAt` differs in this subtle but crucial regard: the name is not past tense, which indicates that it is imperative rather than informative -- "do this at", rather than "this was done at". A better name, suggested by Hidde, would be `reconcile.fluxcd.io/requestedAt`, which has the right connotation, and leaves room for other performatives like `reconcile.fluxcd.io/forcedAt`.

**Comment by hiddeco** [comment]
> A better name, suggested by Hidde, would be `reconcile.fluxcd.io/requestedAt`, which has the right connotation, and leaves room for other performatives like `reconcile.fluxcd.io/forcedAt`.

Another discussion related to this is #103, which would then become `reconcile.fluxcd.io/testedAt`.

**Comment by squaremo** [comment, marked as answer]
Implementation of backward-compatible annotation scheme: https://github.com/fluxcd/pkg/pull/37
